sdk/core/azure-core-amqp/inc/azure/core/amqp/connection.hpp

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-Licence-Identifier: MIT
+
+#pragma once
+
+#include "common/async_operation_queue.hpp"
+#include "connection_string_credential.hpp"
+#include "models/amqp_value.hpp"
+#include "session.hpp"
+#include <azure/core/credentials/credentials.hpp>
+#include <chrono>
+#include <memory>
+#include <string>
+
+namespace Azure { namespace Core { namespace Amqp {
+
+  namespace _detail {
+    class ConnectionImpl;
+  }
+  namespace _internal {
+    class Session;
+
+    struct ConnectionOptions
+    {
+      //    std::chrono::seconds IdleTimeout{std::chrono::seconds(1)};
+      uint32_t MaxFrameSize{512};
+      uint16_t MaxSessions{65535};
+
+      Azure::Core::Amqp::Models::AmqpValue Properties;
+
+      std::chrono::seconds Timeout{0};
```

Comment:
What does 0 as the default timeout mean? Worth a comment.

Reply:
#4504